src/chainstate/stacks/db/blocks.rs

```diff
@@ -3662,10 +3661,152 @@ impl StacksChainState {
         Ok((fees, burns, receipts))
     }
 
+    /// Returns the total number of blocks with Stacking bitcoin ops processed,
+    ///  and the new Stacking bitcoin ops that must be processed in this block.
+    fn get_stacking_ops(
+        clarity_db: &mut ClarityInstance,
+        sort_db: &mut SortitionHandleTx,
+        parent_consensus_hash: &ConsensusHash,
+        parent_block_hash: &BlockHeaderHash,
+        my_consensus_hash: &ConsensusHash,
+    ) -> Result<(u64, Vec<StackStxOp>), Error> {
+        let chain_tip =
+            SortitionDB::get_sortition_id_by_consensus(sort_db.tx(), my_consensus_hash)?
+                .expect("BUG: failed to find SortitionID for current consensus hash");
+        let cur_blocks = sort_db.get_blocks_with_stacks_txs(false, &chain_tip)?;
+
+        let parent_tip = StacksBlockId::new(parent_consensus_hash, parent_block_hash);
+        let mut connection =
+            clarity_db.read_only_connection(&parent_tip, &NULL_HEADER_DB, &NULL_BURN_STATE_DB);
+        let processed_blocks =
+            connection.with_clarity_db_readonly(|db| db.get_stx_btc_ops_processed());
+        connection.done();
+
+        let mut stacking_ops = vec![];
+        // process the interval (processed_block, cur_blocks]
```

Comment:
How big can this range be?  Is there a way to cap it -- i.e. perhaps we can say that this range is no bigger than the maximal number of reward cycles, times the length of a reward cycle?

Reply:
Yes, I think this can be capped at (maximum number of cycles that can be locked for * cycle length)